File: backend/src/main/scala/microservices/order/objects/OrderLineItemPlannerResponse.scala
package com.typesafe.travel.order.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class OrderLineItemPlannerResponse(
    orderItemId: String,
    orderItemKind: String,
    orderItemStatus: String,
    supplierReviewStatus: String,
    bookedAmount: String,
    bookedCurrency: String,
    summaryLabel: String
)
object OrderLineItemPlannerResponse:
  given Encoder[OrderLineItemPlannerResponse] = deriveEncoder
  given Decoder[OrderLineItemPlannerResponse] = deriveDecoder

File: frontend/src/microservices/order/objects/OrderLineItemPlannerResponse.ts
// 本文件定义订单行项目响应，和后端 `OrderLineItemPlannerResponse` 保持同名镜像。

export type OrderLineItemPlannerResponse = {
  orderItemId: string
  orderItemKind: string
  orderItemStatus: string
  supplierReviewStatus: string
  bookedAmount: string
  bookedCurrency: string
  summaryLabel: string
}

export const orderLineItemPlannerResponseFromJson = (json: string): OrderLineItemPlannerResponse =>
  JSON.parse(json) as OrderLineItemPlannerResponse

export const orderLineItemPlannerResponseToJson = (value: OrderLineItemPlannerResponse): string =>
  JSON.stringify(value)
